coral/src/app/features/user-information/tenant-information/TenantInformation.tsx

```diff
@@ -0,0 +1,56 @@
+import { Alert, Box, Input } from "@aivenio/aquarium";
+import { useQuery } from "@tanstack/react-query";
+import { TenantInformationData } from "src/app/features/user-information/tenant-information/TenantInformationData";
+import { TenantInformationForm } from "src/app/features/user-information/tenant-information/TenantInformationForm";
+import { getMyTenantInfo } from "src/domain/user/user-api";
+import { parseErrorMsg } from "src/services/mutation-utils";
+
+const TenantInformation = () => {
+  const { data, isError, error } = useQuery(
+    ["getMyTenantInfo"],
+    getMyTenantInfo
+  );
+
+  if (data === undefined) {
+    return (
+      <>
+        {isError && <Alert type="error">{parseErrorMsg(error)}</Alert>}
+        {!isError && (
```

Comment:
This is for the loading state, right? We should add information for screen reader users, too. 

adding a div or span with class `visually-hidden` and something like "Loading..."

Reply:
Also nitpick: I think we usually use `isLoading` do determine the loading state, which I feel is more readable in the code. 

- if isLoading -> return this
- if isError -> return this
- else -> return this

Reply:
> adding a div or span with class visually-hidden and something like "Loading..."

The new aquarium version actually has `Loading...` as `aria-label`! https://github.com/aiven/design-system/blob/693ba7e96dab653c39c0dc82100fa91e242a0265/src/molecules/Skeleton/Skeleton.tsx#L58

They are actively working on fixing a11y issues flagged by `axe` in their CI.

>  I think we usually use isLoading do determine the loading state,

It's a tradeoff, I think: either we use `isLoading`, but then `data` is type as `Thing | undefined`, which is a pain, and false, or we use `data === undefined`, which is less semantically significant, but allows `data` to be typed properly when we render stuff.

Reply:
Ah nice, I didn't notice that skeleton now has that info! 

>It's a tradeoff, I think: either we use isLoading, but then data is always typed as Thing | undefined, which is a pain, and false, or we use data === undefined, which is less semantically significant, but allows data to be typed properly when we render stuff.

I _think_ (and you have to check, but I had that in `Profile.tsx` that when `isLoading` and `isError` return something explicit, then TS can understand that `data` from useQuery is always defined (if that's the case in typing of the api of course), for example:

```

if (isLoading) return <LoadingInfo>

if(isError) return <ErrorStuff> 

return <ActualComponent>

```

But it doesn't really matter, since both works!

Reply:
Whoa. So it only works if there is an `isLoading` AND and `isError` check beforehand. That is much smarter than I expected TS to be.. So probably the pattern we should follow is something like this:

```tsx
const TenantInformation = () => {
  const { data, isLoading, isError, error } = useQuery(
    ["getMyTenantInfo"],
    getMyTenantInfo
  );

  if (isLoading) {
    return (
      <Box.Flex flexDirection="column" maxWidth={"lg"}>
        <Box.Flex flexDirection="column">
          <Input.Skeleton />
        </Box.Flex>
        <Box.Flex flexDirection="column">
          <Input.Skeleton />
        </Box.Flex>
        <Box.Flex flexDirection="column">
          <Input.Skeleton />
        </Box.Flex>
      </Box.Flex>
    );
  }

  if (isError) {
    return <Alert type="error">{parseErrorMsg(error)}</Alert>;
  }

  return (
    <TenantInformationData
      tenantName={data.tenantName}
      orgName={data.orgName}
      contactPerson={data.contactPerson}
      description={data.tenantDesc}
    />
  );
};
```

Reply:
Actually can be even cleaner, no need for the boxes in the loading state: [7b684fa](https://github.com/Aiven-Open/klaw/pull/2058/commits/7b684fad2968a37a6a3e0dff2f57b72f217c3c07)

Reply:
Noice!